### ISSUE TITLE
Mask Extension does not seem to work with HeatmapLayer [Bug]

### ISSUE THREAD

**Issue by andresramosp** [visgl/deck.gl#7184]
### Description

I took the official example for HeatmapLayer (https://github.com/visgl/deck.gl/tree/8.8-release/examples/website/heatmap), and just changed the dataset and added a new GeoJsonLayer with a polygon to mask the heatmap. 

This image shows the GeoJsonLayer and the HeatmapLayer without applying the mask extension yet:

![image](https://user-images.githubusercontent.com/44118551/183243799-ea285fcb-97cf-4979-b215-4aa5341e09dc.png)

Then, when applying the mask extension, the first rendering just shows the heatmap without mask:, 

![image](https://user-images.githubusercontent.com/44118551/183243830-dcfc3a85-160c-4a16-811b-53b626cf59ae.png)

and when I zoom the map, the heatmap dissapears and I got the following warning on console:

[.WebGL-00002132058C4E00] GL_INVALID_OPERATION: Feedback loop formed between Framebuffer and active Texture.

If I replace the HeatmapLayer by a ScatterplotLayer, all works as expected:

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

The heatmap clipped by the polygon mask.

### Steps to Reproduce

https://github.com/andresramosp/DeckGL_Heatmap_Masked

### Environment

- Framework version: 8.8.0
- Browser: Chrome
- OS: Windows

### Logs

[.WebGL-00002132058C4E00] GL_INVALID_OPERATION: Feedback loop formed between Framebuffer and active Texture.

**Comment by sliterok** [comment]
I have same problem with ScenegraphLayer. Is it expected behavior?

**Comment by felixpalmer** [comment]
@sliterok I don't think it should be related. Could you make a separate issue with a jsfiddle demonstrating your issue?

**Comment by andresramosp** [comment]
Hi Felix! Any estimate for the fix for Mask on HeatmapLayer? Did you figure out the problem?
